WalletWasabi.Gui/ViewModels/StatusBarViewModel.cs

```diff
@@ -138,26 +144,34 @@ public StatusBarViewModel(NodesCollection nodes, WasabiSynchronizer synchronizer
 				RefreshStatus();
 			});
 
-			this.WhenAnyValue(x => x.Status).Subscribe(async status =>
+			this.WhenAnyValue(x => x.Status).Subscribe(status =>
 			{
-				if (status.EndsWith(".")) // Then do animation.
+				StatusText = StatusBarStatusStringConverter.Convert(status);
+			});
+
+			this.WhenAnyValue(x => x.StatusText).Subscribe(async status =>
+			{
+				using (await StatusTextAnimationLock.LockAsync()) // Without this lock the status get stuck once in a while.
 				{
-					string nextAnimation = null;
-					if (status.EndsWith("..."))
-					{
-						nextAnimation = status.TrimEnd("..", StringComparison.Ordinal);
-					}
-					else if (status.EndsWith("..") || status.EndsWith("."))
+					if (status.EndsWith(".")) // Then do animation.
 					{
-						nextAnimation = $"{status}.";
-					}
+						string nextAnimation = null;
+						if (status.EndsWith("..."))
+						{
+							nextAnimation = status.TrimEnd("..", StringComparison.Ordinal);
+						}
+						else if (status.EndsWith("..") || status.EndsWith("."))
```

Comment:
This comment is a bit silly but, you can remove the `status.EndsWith("..")` here.

Reply:
ACK